src/components/layout/errorPage.tsx

```diff
@@ -0,0 +1,91 @@
+import type { NextPage } from 'next';
+import { NextSeo } from 'next-seo';
+import { useRouter } from 'next/router';
+
+import CustomImage from '../decoration/customImage';
+import { DarkButton } from '../decoration/buttons';
+import useErrorStore from '../../lib/stores/errorStore';
+
+interface ErrorProps {
+  statusCode?: number;
+}
+
+const Error: NextPage<ErrorProps> = ({
+  statusCode: statusCodeProp,
```

Comment:
Instead of doing this and then conditionally changing a new variable, you can add a default value to the prop
```suggestion
  statusCode = 404,
```